### PR TITLE
offloading microcycles (Hess Op 3 and constructor)

### PR DESCRIPTION
**Pull request by valay1**

### CODE REVIEW

File: my_pyscf/mcscf/lasci_sync.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def kernel (las, mo_coeff=None, ci0=None, casdm0_fr=None, conv_tol_grad=1e-4,
     log = lib.logger.new_logger(las, verbose)
     t0 = (lib.logger.process_clock(), lib.logger.perf_counter())
     log.debug('Start LASCI')
-
+    gpu=las.use_gpu
```
Comment (MatthewRHermes) [inline my_pyscf/mcscf/lasci_sync.py:30]: what is the point of this?

File: my_pyscf/mcscf/lasci_sync.py [hunk 2]
```diff
@@ -1106,33 +1110,38 @@ def get_veff (self, dm1s_mo=None):
...
-            t1 = lib.logger.timer (self.las, 'vk_mo (bb|jj) in microcycle', *t1)   
-            t1 = lib.logger.timer (self.las, 'vk_mo (bi|aj) in microcycle', *t1);
-
-        else: 
+            dm1_ao=np.dot(mo,np.dot(dm1_mo,moH))
+            veff_ao=np.squeeze(self.las.get_veff(dm1s=dm1_ao))
+            return np.dot(moH,np.dot(veff_ao,mo))
+        else:
+            if getattr (self, 'bPpj', None) is None:
+                dm1_ao = np.dot (mo, np.dot (dm1_mo, moH))
```
Comment (MatthewRHermes) [inline my_pyscf/mcscf/lasci_sync.py:1119]: 1119-1121 is the same as 1114-1116. Why not put it in the same conditional block? Why indent everything? ("Return" ends the function execution; a "return" at the end of an "if" block means you don't need the "else" block)

Comment (valay1) [reply my_pyscf/mcscf/lasci_sync.py]: OK I made a cleaner block on my fork. Do I just open another pull request?

**Comment by MatthewRHermes** [comment]
To resolve the conflicts, first pull my gpu branch, then merge with your valay_gpu branch locally, resolve the conflicts yourself, commit and push to your remote valay_gpu

File: my_pyscf/mcscf/lasci_sync.py (hunk 2 above)
Comment (MatthewRHermes) [reply my_pyscf/mcscf/lasci_sync.py]: This PR will track any update you push to valay1:valay_gpu as long as it stays open.

**Pull request merged**